e2e/testcases/reconciler_manager_test.go

```diff
@@ -603,3 +603,38 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	})
 
 }
+
+func TestAllowVerticalScaleResourceAdjustment(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo)
+
+	reconcilerDeployment := &appsv1.Deployment{}
+	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, reconcilerDeployment); err != nil {
+		nt.T.Fatal(err)
+	}
+	rs := &v1beta1.RootSync{}
+	if err := nt.Get(configsync.RootSyncName, configsync.ControllerNamespace, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// allow vertical scale
+	_, err := nt.Kubectl("patch", "deployment", "reconciler-manager", "-n", "config-management-system", "-p", "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"args\":[\"--enable-leader-election\", \"--allow-vertical-scale\"],\"name\":\"reconciler-manager\"}]}}}}")
+	if err != nil {
+		nt.T.Fatalf("Error enabling vertical scale in deployment", err)
+	}
+
+	// update numbers are compliant with autopilot requirement
+	var updatedFirstContainerCPURequest int64 = 430
+	var updatedFirstContainerMemoryRequest int64 = 200
+
+	// simulating VPA update to deployment
+	_, err = nt.Kubectl("patch", "deployment", "root-reconciler", "-n", "config-management-system", "-p", fmt.Sprintf("{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"%s\",\"resources\":{\"requests\":{\"cpu\":\"%dm\",\"memory\":\"%dMi\"}}}]}}}}", "hydration-controller", updatedFirstContainerCPURequest, updatedFirstContainerMemoryRequest))
+	if err != nil {
+		nt.T.Fatalf("Error simulating VPA deployment update", err)
+	}
+
+	nomostest.Wait(nt.T, "Waiting for patch deployment update", nt.DefaultWaitTimeout, func() error {
+		return nt.Validate("root-reconciler", v1.NSConfigManagementSystem, &appsv1.Deployment{},
+			hasGeneration(2), firstContainerCPURequestIs(updatedFirstContainerCPURequest),
```

Comment:
in the previous commit, the test was updating `reconciler-manager` deployment. Since there was a --enable-vertical-scale flag addition (+1) and container resource update (+1), the expected generation is 3. The test has been modified to update `hydration-controller` container in `root-reconciler` deployment instead, so there's only resource update (+1) therefore the expected deployment is 2. After this feedback, we add `minReadySeconds` update (+1) that should be reverted (+1), so the expected deployment is now 3

Reply:
unresolving the comments for now, let me know if this makes sense

Reply:
ack